File: official/docs/csharp/v6/smartrate/retrieve-recommend-ship-date.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            EasyPost.Parameters.Shipment.RecommendShipDateForShipment parameters = new()
            {
                DesiredDeliveryDate = "2024-07-18",
            };

            List<EasyPost.Models.API.RecommendShipDateForShipmentResult> rates = await client.Shipment.RecommendShipDate("shp_...", parameters);

            Console.WriteLine(JsonConvert.SerializeObject(rates, Formatting.Indented));
        }
    }
}

File: official/docs/golang/v4/smartrate/retrieve-recommend-ship-date.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v4"
)

func retrieveRecommendShipDate() {
	client := easypost.New("EASYPOST_API_KEY")

	rates, _ := client.RecommendShipDateForShipment("shp_...", "YYYY-MM-DD")

	fmt.Println(rates)
}
